hail/src/main/scala/is/hail/expr/ir/orderings/ShuffleOrdering.scala

```diff
@@ -19,7 +19,7 @@ object ShuffleOrdering {
             val bcode1 = x.asInstanceOf[SCanonicalShufflePointerCode].binaryRepr
             val bcode2 = y.asInstanceOf[SCanonicalShufflePointerCode].binaryRepr
             val ord = BinaryOrdering.make(bcode1.st, bcode2.st, ecb)
-            ord.compareNonnull(cb, x.asString.asBytes(), y.asString.asBytes())
+            ord.compareNonnull(cb, x.asInstanceOf[SCanonicalShufflePointerCode].binaryRepr, y.asInstanceOf[SCanonicalShufflePointerCode].binaryRepr)
```

Comment:
can't this just use `bcode1` and `bcode2`?

Reply:
oh, yeah.